aws/client.go

```diff
@@ -0,0 +1,148 @@
+package aws
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+
+	"github.com/quay/alas"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+)
+
+const (
+	amazonLinux1Mirrors = "http://repo.us-west-2.amazonaws.com/2018.03/updates/x86_64/mirror.list"
+	amazonLinux2Mirrors = "https://cdn.amazonlinux.com/2/core/latest/x86_64/mirror.list"
+	repoDataPath        = "/repodata/repomd.xml"
+	updatesPath         = "/repodata/updateinfo.xml.gz"
+)
+
+var logger = log.With().Str("component", "aws-alas-client").Logger()
+
+// Client is an http for accessing ALAS mirrors.
+type Client struct {
+	c       *http.Client
+	mirrors []*url.URL
+	logger  zerolog.Logger
+}
+
+func NewClient(release Release) (*Client, error) {
+	client := &Client{
+		c:       &http.Client{},
+		mirrors: []*url.URL{},
+		logger:  log.With().Str("component", "aws-alas-client").Str("release", string(release)).Logger(),
+	}
+	err := client.getMirrors(release)
+	return client, err
+}
+
+// RepoMD returns a alas.RepoMD containing sha256 information of a repositories contents
+func (c *Client) RepoMD() (alas.RepoMD, error) {
+	for _, mirror := range c.mirrors {
+		m := *mirror
+		m.Path = path.Join(m.Path, repoDataPath)
```

Comment:
If this is a url.URL, you can use `.Parse()` to resolve the reference.

Reply:
Going to skip on this one, amazon returns urls which do not end in a slash so the u.Parse() does not to the right thing. https://play.golang.org/p/Uiy1zWJh4sY

I don't want to add a "/" to the end of each string, id prefer just the path.Join method